stores/data.ts

```diff
@@ -0,0 +1,140 @@
+import { defineStore } from "pinia";
+import { ref } from "vue";
+import { VALID_COMMUNITIES, isValidCommunity } from "~/utils/luts";
+
+export interface LandslideData {
+  expires_at: string;
+  precipitation_24hr: number;
+  precipitation_2days: number;
+  precipitation_3days: number;
+  precipitation_inches: number;
+  precipitation_mm: number;
+  risk_24hr: number;
+  risk_2days: number;
+  risk_3days: number;
+  risk_is_elevated_from_previous: boolean;
+  risk_level: number;
+  timestamp: string;
+}
+
+export const useDataStore = defineStore("data", () => {
+  const data = ref<LandslideData | null>(null);
+  const loading = ref<boolean>(false);
+  const error = ref<string | null>(null);
+
+  // Get API URL from Nuxt runtime config
+  const { $config } = useNuxtApp();
+  const apiUrl = $config.public.snapApiUrl;
+
+  const formatTimeDifference = (diffMs: number): string => {
+    const diffMins = Math.floor(diffMs / 60000);
+
+    if (diffMins < 1) {
```

Comment:
Consider adding [a library like this](https://www.npmjs.com/package/javascript-time-ago) instead.